hip-lang-config.cmake.in

```diff
@@ -68,24 +68,39 @@ include( "${CMAKE_CURRENT_LIST_DIR}/hip-lang-targets.cmake" )
 #get_filename_component cannot resolve the symlinks if called from /opt/rocm/lib/hip
 #and do three level up again
 get_filename_component(_DIR "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
+# root directory of the hip library installation
 get_filename_component(_IMPORT_PREFIX "${_DIR}/../../../" REALPATH)
 
+# extract HIP_CLANG_ROOT from CMAKE_HIP_COMPILER
+find_program(CMAKE_HIP_COMPILER_WITH_PATH ${CMAKE_HIP_COMPILER})
+cmake_path(GET CMAKE_HIP_COMPILER_WITH_PATH PARENT_PATH HIP_CLANG_BIN)
+cmake_path(GET HIP_CLANG_BIN PARENT_PATH HIP_CLANG_ROOT)
 
-#need _IMPORT_PREFIX to be set
-file(GLOB HIP_CLANG_INCLUDE_SEARCH_PATHS "${_IMPORT_PREFIX}/../llvm/lib/clang/*/include")
+#need HIP_CLANG_ROOT to be set
+file(GLOB HIP_CLANG_INCLUDE_SEARCH_PATHS "${HIP_CLANG_ROOT}/lib/clang/*/include")
 find_path(HIP_CLANG_INCLUDE_PATH __clang_cuda_math.h
     HINTS ${HIP_CLANG_INCLUDE_SEARCH_PATHS}
     NO_DEFAULT_PATH)
 get_filename_component(HIP_CLANG_INCLUDE_PATH "${HIP_CLANG_INCLUDE_PATH}" DIRECTORY)
 
-#If HIP isnot installed under ROCm, need this to find HSA assuming HSA is under ROCm
+file(GLOB HIP_CLANGRT_LIB_SEARCH_PATHS "${HIP_CLANG_ROOT}/lib/clang/*/lib/*")
+find_library(CLANGRT_BUILTINS
+    NAMES
+      clang_rt.builtins
+      clang_rt.builtins-x86_64
+    PATHS
+      ${HIP_CLANGRT_LIB_SEARCH_PATHS}
+      ${HIP_CLANG_INCLUDE_PATH}/../lib/linux)
+
+#If HIP is not installed under ROCm, need this to find HSA assuming HSA is under ROCm
 if( DEFINED ENV{ROCM_PATH} )
   set(ROCM_PATH "$ENV{ROCM_PATH}")
 endif()
 
 #if HSA is not under ROCm then provide CMAKE_PREFIX_PATH=<HSA_PATH>
 find_path(HSA_HEADER hsa/hsa.h
   PATHS
+    "${CMAKE_HIP_COMPILER_ROCM_ROOT}/include"
```

Comment:
`CMAKE_HIP_COMPILER_ROCM_ROOT` is provided by CMake. Here is the searching order: compiler, next to hip runtime library, ROCM_PATH and then `/opt/rocm`